### PR TITLE
Fixes #3768

### LINKED ISSUE #3768: SubstanceGroup output doesn't correctly quote " symbols
**Describe the bug**
Here's the demonstration:
```
In [2]: m = Chem.MolFromMolBlock('''foo
   ...:      RDKit          2D
   ...: 
   ...:   0  0  0  0  0  0  0  0  0  0999 V3000
   ...: M  V30 BEGIN CTAB
   ...: M  V30 COUNTS 6 6 1 0 0
   ...: M  V30 BEGIN ATOM
   ...: M  V30 1 C -0.866400 -0.499400 0.000000 0
   ...: M  V30 2 C -0.865600 0.500600 0.000000 0
   ...: M  V30 3 C 0.000800 1.000000 0.000000 0
   ...: M  V30 4 C 0.866400 0.499400 0.000000 0
   ...: M  V30 5 C 0.865600 -0.500600 0.000000 0
   ...: M  V30 6 C -0.000800 -1.000000 0.000000 0
   ...: M  V30 END ATOM
   ...: M  V30 BEGIN BOND
   ...: M  V30 1 2 1 2
   ...: M  V30 2 1 2 3
   ...: M  V30 3 2 3 4
   ...: M  V30 4 1 4 5
   ...: M  V30 5 2 5 6
   ...: M  V30 6 1 1 6
   ...: M  V30 END BOND
   ...: M  V30 BEGIN SGROUP
   ...: M  V30 1 DAT 0 ATOMS=(1 1) FIELDNAME=[DUP]Tempstruct FIELDINFO="""" -
   ...: M  V30 FIELDDISP="    2.1037    1.5400    DA    ALL  0       0" QUERYOP="""" -
   ...: M  V30 FIELDDATA=Foo1
   ...: M  V30 END SGROUP
   ...: M  V30 END CTAB
   ...: M  END
   ...: ''')

In [3]: print(Chem.MolToV3KMolBlock(m))
foo
     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 6 6 1 0 0
M  V30 BEGIN ATOM
M  V30 1 C -0.866400 -0.499400 0.000000 0
M  V30 2 C -0.865600 0.500600 0.000000 0
M  V30 3 C 0.000800 1.000000 0.000000 0
M  V30 4 C 0.866400 0.499400 0.000000 0
M  V30 5 C 0.865600 -0.500600 0.000000 0
M  V30 6 C -0.000800 -1.000000 0.000000 0
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 2 1 2
M  V30 2 1 2 3
M  V30 3 2 3 4
M  V30 4 1 4 5
M  V30 5 2 5 6
M  V30 6 1 1 6
M  V30 END BOND
M  V30 BEGIN SGROUP
M  V30 1 DAT 0 ATOMS=(1 1) FIELDNAME=[DUP]Tempstruct FIELDINFO=" FIELDDISP="   -
M  V30  2.1037    1.5400    DA    ALL  0       0" QUERYOP=" FIELDDATA="Foo1"
M  V30 END SGROUP
M  V30 END CTAB
M  END
```

we can see that it's not a parsing problem:
```
In [4]: sgs = Chem.GetMolSubstanceGroups(m)

In [5]: sgs[0].GetPropsAsDict()
Out[5]: 
{'TYPE': 'DAT',
 'index': 1,
 'FIELDNAME': '[DUP]Tempstruct',
 'FIELDINFO': '"',
 'FIELDDISP': '    2.1037    1.5400    DA    ALL  0       0',
 'QUERYOP': '"',
 'DATAFIELDS': <rdkit.rdBase._vectNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE at 0x7f334d06c990>}
```

**Expected behavior**
`"` characters in SubstanceGroup output need to be enclosed in `"` and doubled.

**Configuration (please complete the following information):**
 - RDKit version: master and releases
 - OS: all

### PR DESCRIPTION
**Pull request by greglandrum**
simple change to make sure we double double quotes in the output of V3K mol blocks

### CODE REVIEW

File: Code/GraphMol/FileParsers/MolSGroupWriting.cpp [hunk 1]
```diff
@@ -511,15 +511,22 @@ std::string FormatV3000StringPropertyBlock(const std::string &prop,
   if (sgroup.getPropIfPresent(prop, propValue)) {
     if (!propValue.empty()) {
       ret << ' ' << prop << '=';
-      bool hasSpaces = propValue.find(' ') != std::string::npos;
+      bool needsQuotes = propValue.find(' ') != std::string::npos ||
+                         propValue.find('"') != std::string::npos;
```
Comment (ricrogz) [inline Code/GraphMol/FileParsers/MolSGroupWriting.cpp:515]: This check for double quotes is broader than what the specification says, but I think it should not be a problem (enclosing double quotes will be stripped away at parsing):
image

I think it is also a small change, but would you mind adding the case of the string starting with an opening parenthesis?

**Review by ricrogz — CHANGES_REQUESTED** [review]
Looks good, just a small request.

File: Code/GraphMol/FileParsers/MolSGroupWriting.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/FileParsers/MolSGroupWriting.cpp:515]: stupid me for applying logic to the problem ;-)
I will implement what is in the spec.

Comment (greglandrum) [reply Code/GraphMol/FileParsers/MolSGroupWriting.cpp:515]: ok, I did a bit more investigation and it turns out that neither our parser nor BIOVIA Draw will correctly parse `FIELDINFO=foo""`, but both are fine with `FIELDINFO="foo"""`. 
BIOVIA Draw also parses `FIELDINFO=foo(bar)` as just `foo`.
BIOVIA Draw and Marvin Sketch are both fine with fields that don't need to be quoted starting with double quotes

So I think we should always include the (theoretically) extra starting double quotes for cases where the property includes either a double quote or a left paren.

I will update the PR to recognize the case with the paren as well.

File: Code/GraphMol/FileParsers/file_parsers_catch.cpp [hunk 2]
```diff
@@ -2931,4 +2931,74 @@ TEST_CASE("supplier close methods") {
...
+  //     auto sgs = getSubstanceGroups(*m);
+  //     REQUIRE(sgs.size() == 1);
+  //     auto sg = sgs[0];
+  //     CHECK(sg.getProp<std::string>("FIELDINFO") == "\"");
+  //     CHECK(sg.getProp<std::string>("QUERYOP") == "\"\"");
+  //     auto mb = MolToV3KMolBlock(*m);
+  //     std::cerr << mb << std::endl;
+  //     CHECK(mb.find("FIELDINFO=\"\"\"\"") != std::string::npos);
+  //     CHECK(mb.find("QUERYOP=\"\"\"\"\"") != std::string::npos);
+  //   }
```
Comment (ricrogz) [inline Code/GraphMol/FileParsers/file_parsers_catch.cpp:2971]: Should this test be uncommented? (or just removed?)

**Review by ricrogz — APPROVED** [review]
Looks good now. Just wondering about the test that's commented out.

**Pull request merged**